op-batcher/batcher/channel_builder_test.go

```diff
@@ -618,18 +529,51 @@ func ChannelBuilder_OutputFramesMaxFrameIndex(t *testing.T, batchType uint) {
 	require.ErrorIs(t, cb.FullErr(), ErrMaxFrameIndex)
 }
 
-// ChannelBuilder_AddBlock tests the AddBlock function
+// TestChannelBuilder_FullShadowCompressor is a regression test testing that
+// the shadow compressor is correctly marked as full if adding another block
+// would produce a leftover frame.
+//
+// This test fails in multiple places if the subtraction of
+// [derive.FrameV0OverHeadSize] in [MaxDataSize] is omitted, which has been the
+// case before it got fixed it #9887.
+func TestChannelBuilder_FullShadowCompressor(t *testing.T) {
+	require := require.New(t)
+	cfg := ChannelConfig{
+		MaxFrameSize:    752,
+		TargetNumFrames: 1,
+		BatchType:       derive.SpanBatchType,
+	}
+	cfg.InitShadowCompressor()
+
+	cb, err := NewChannelBuilder(cfg, defaultTestRollupConfig, latestL1BlockOrigin)
+	require.NoError(err)
+
+	rng := rand.New(rand.NewSource(420))
+	a := dtest.RandomL2BlockWithChainId(rng, 1, defaultTestRollupConfig.L2ChainID)
+	_, err = cb.AddBlock(a)
+	require.NoError(err)
+	_, err = cb.AddBlock(a)
+	require.ErrorIs(err, derive.CompressorFullErr) // would fail without fix, would be nil
+	// without fix, adding a third block would fail with full error
```

Comment:
also *with* fix, adding a third block would fail with full error, right?

Reply:
yes, but adding the 2nd block already fails with fix. might change this to 
```suggestion
	// without fix, only adding a third block would fail with full error
```